packages/plugin-core/src/telemetry.ts

```diff
@@ -0,0 +1,44 @@
+import { SegmentClient } from "@dendronhq/common-server";
+import * as vscode from "vscode";
+import { Logger } from "./logger";
+import { DendronWorkspace } from "./workspace";
+import _ from "lodash";
+
+export function isVSCodeTelemetryEnabled(): boolean | undefined {
+  // `isTelemetryEnabled` only seems to be available on VSCode 1.55 and above.
+  // Our dependencies are currently set up for 1.54, so typescript does not understand that it exists.
+  try {
+    return (vscode.env as any).isTelemetryEnabled;
+  } catch (err) {
+    Logger.error({ msg: "Error when trying to access telemetry preference" });
+    return undefined;
+  }
+}
+
+/** Creates a SegmentClient for telemetry, if enabled, and listens for vscode telemetry settings to disable it when requested. */
+export function setupSegmentClient(ws: DendronWorkspace) {
+  function instantiateSegmentClient() {
+    SegmentClient.instance({
+      optOut: ws.config.noTelemetry || !isVSCodeTelemetryEnabled(),
```

Comment:
If I'm not misunderstanding you, I don't think that's a good idea. Then, when the user enabled telemetry in VSCode settings, we won't know if they manually disabled Dendron telemetry or if it was because they had the VSCode setting disabled before.

My main concern is that if a user has never touched "Dendron: Enable Telemetry" and "Dendron: Disable Telemetry" commands, then Dendron should always follow the VSCode settings. If they did, then that takes precedence.

Reply:
that's fair. one thing we should be able to track though is number of workspaces with telemetry disabled. this tells us the delta in our metrics. so whether its calling `DisableTelemetry` or calling `    AnalyticsUtils.track(VSCodeEvents.DisableTelemetry);` we should know how many workspaces have telemetry disabled